aptos-move/framework/src/natives/cryptography/ed25519.rs

```diff
@@ -131,123 +120,31 @@ fn native_ed25519_verify(
-#[derive(Debug, Clone)]
-pub struct Secp256k1ECDSARecoverGasParameters {
-    pub base_cost: InternalGas,
-}
-
-fn native_secp256k1_ecdsa_recover(
-    gas_params: &Secp256k1ECDSARecoverGasParameters,
-    _context: &mut NativeContext,
-    _ty_args: Vec<Type>,
-    mut arguments: VecDeque<Value>,
-) -> PartialVMResult<NativeResult> {
-    debug_assert!(_ty_args.is_empty());
-    debug_assert!(arguments.len() == 3);
-
-    let signature = pop_arg!(arguments, Vec<u8>);
-    let recovery_id = pop_arg!(arguments, u8);
-    let msg = pop_arg!(arguments, Vec<u8>);
-
-    let cost = gas_params.base_cost;
-
-    // NOTE(Gas): O(1) cost
-    // (In reality, O(|msg|) deserialization cost, with |msg| < libsecp256k1_core::util::MESSAGE_SIZE
-    // which seems to be 32 bytes, so O(1) cost for all intents and purposes.)
-    let msg = match libsecp256k1::Message::parse_slice(&msg) {
-        Ok(msg) => msg,
-        Err(_) => {
-            return Ok(NativeResult::ok(
-                cost,
-                smallvec![Value::vector_u8([0u8; 0]), Value::bool(false)],
-            ));
-        }
-    };
-
-    // NOTE(Gas): O(1) cost
-    let rid = match libsecp256k1::RecoveryId::parse(recovery_id) {
-        Ok(rid) => rid,
-        Err(_) => {
-            return Ok(NativeResult::ok(
-                cost,
-                smallvec![Value::vector_u8([0u8; 0]), Value::bool(false)],
-            ));
-        }
-    };
-
-    // NOTE(Gas): O(1) deserialization cost
-    // which seems to be 64 bytes, so O(1) cost for all intents and purposes.
-    let sig = match libsecp256k1::Signature::parse_standard_slice(&signature) {
-        Ok(sig) => sig,
-        Err(_) => {
-            return Ok(NativeResult::ok(
-                cost,
-                smallvec![Value::vector_u8([0u8; 0]), Value::bool(false)],
-            ));
-        }
-    };
-
-    // NOTE(Gas): O(1) cost: a size-2 multi-scalar multiplication
-    let pk = match libsecp256k1::recover(&msg, &sig, &rid) {
-        Ok(pk) => pk,
-        Err(_) => {
-            return Ok(NativeResult::ok(
-                cost,
-                smallvec![Value::vector_u8([0u8; 0]), Value::bool(false)],
-            ));
-        }
-    };
-
-    Ok(NativeResult::ok(
-        cost,
-        smallvec![
-            Value::vector_u8(pk.serialize()[1..].to_vec()),
-            Value::bool(true)
-        ],
-    ))
-}
-
 /***************************************************************************************************
  * module
  *
  **************************************************************************************************/
 #[derive(Debug, Clone)]
 pub struct GasParameters {
-    // Ed25519 signatures
-    pub ed25519_validate_pubkey: Ed25519ValidatePubkeyGasParameters,
-    pub ed25519_verify: Ed25519VerifyGasParameters,
-
-    // ECDSA signatures based on secp256k1 elliptic curves
-    pub secp256k1_ecdsa_recover: Secp256k1ECDSARecoverGasParameters,
+    pub base_cost: InternalGas,
```

Comment:
Are you sure you don't need a different base cost for different functions?

Reply:
As far as I can tell, all of these functions do mostly cryptographic work, so using the same base cost is reasonable. @davidiw.

Reply:
...and from discussing with you, Victor.